ros/angel_utils/nodes/simple_2d_detection_overlay.cxx

```diff
@@ -406,6 +423,129 @@ ::collect_detections( ObjectDetection2dSet::SharedPtr const det_set )
   ++m_detset_count;
 }
 
+// ----------------------------------------------------------------------------
+void
+Simple2dDetectionOverlay
+::collect_joints( HandJointPosesUpdate::SharedPtr const joints_msg )
+{
+  auto log = this->get_logger();
+
+  // lookup image for joints_msg->source_stamp
+  // check that detection header frame_id matches
+  size_t source_nanosec_key =
+    time_key_from_header( joints_msg->source_stamp );
+  RCLCPP_DEBUG( log, "Joint source key: %zu", source_nanosec_key );
+
+  // Check that we have the image in our history that matches our joints
+  // received.
+  auto find_it = m_frame_map.find( source_nanosec_key );
+  if( find_it == m_frame_map.end() )
+  {
+    auto history_min_key = m_frame_map.begin()->first;
+    RCLCPP_WARN(
+      log,
+      "Failed to find an image in our history that matches received joints. "
+      "joint source (%zu) < (%zu) history min.",
+      source_nanosec_key, history_min_key
+      );
+    return;
+  }
+
+  // Found something. Double-checking frame source is set to the same thing?
+  if( joints_msg->header.frame_id != find_it->second->header.frame_id )
+  {
+    RCLCPP_WARN(
+      log,
+      "Received joints frame-id does not match the aligned image in "
+      "our history? d-set (%s) != (%s) image",
+      joints_msg->header.frame_id.c_str(),
+      find_it->second->header.frame_id.c_str()
+      );
+    return;
+  }
+
+  // Only plot the patient skeleton
+  if(joints_msg->hand != "patient")
+  {
+    return;
+  }
+
+  // Make a copy of the message image to plot on.
+  cv_bridge::CvImagePtr img_ptr =
+    cv_bridge::toCvCopy( find_it->second, "rgb8" );
+
+  std::map<std::string, std::vector<double>> joint_positions = {};
+  std::map<std::string, std::vector<std::string>> connection_lines = { // mapping of joint line connections
+    {"nose", {"mouth"}},
+    {"mouth", {"throat"}},
+    {"throat", {"chest", "left_upper_arm", "right_upper_arm"}},
+    {"chest", {"back"}},
+    {"left_upper_arm", {"left_lower_arm"}},
+    {"left_lower_arm", {"left_wrist"}},
+    {"left_wrist", {"left_hand"}},
+    {"right_upper_arm", {"right_lower_arm"}},
+    {"right_lower_arm", {"right_wrist"}},
+    {"right_wrist", {"right_hand"}},
+    {"back", {"left_upper_leg", "right_upper_leg"}},
+    {"left_upper_leg", {"left_knee"}},
+    {"left_knee", {"left_lower_leg"}},
+    {"left_lower_leg", {"left_foot"}},
+    {"right_upper_leg", {"right_knee"}},
+    {"right_knee", {"right_lower_leg"}},
+    {"right_lower_leg", {"right_foot"}}
+  };
+
+  static auto const COLOR_PT = cv::Scalar{ 0, 255, 0 }; 
```

Comment:
This could probably be defined at the top of the file with the other statics, but here is fine unless we want to reuse this or coordinate with other colors for other overlays.